src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -343,7 +345,43 @@ impl<'a, 'tcx> GatherLoanCtxt<'a, 'tcx> {
         // Create the loan record (if needed).
         let loan = match restr {
             RestrictionResult::Safe => {
-                // No restrictions---no loan record necessary
+                let loan_scope = match *loan_region {
```

Comment:
I should have moved this and the below `match` expression into its own function. I'll take care of that.